File: 2020/Day13/Part1.py
file = open("Input.txt", "r")
f = file.read()
lines = f.split('\n')
timeStamp = int(lines[0])
mm = timeStamp
mn = 0
buses = lines[1].split(',')
for val in buses:
    if val != 'x':
        num = int(val)
        m = num - (timeStamp % num)
        if m < mm:
            mm = m
            mn = num

print(mn,mm,mm*mn)

File: 2021/Days/Day07/Part1.hs
module Main where
import System.IO 
import Data.List.Split
import Debug.Trace
import Data.List



main :: IO ()
main = do
    contents <- readFile "Days/Day07/Input.txt" 
    let startingPositions = sort (map readInt (splitOn "," contents))
    let halfLength = length startingPositions `div` 2
    let firstHalf = take halfLength startingPositions
    let secondHalf = take halfLength (reverse startingPositions)
    print (getFuelCost firstHalf secondHalf)

readInt :: String -> Int
readInt s = read s :: Int

getFuelCost :: [Int] -> [Int] -> Int 
getFuelCost smaller larger = sum (zipWith (-) larger smaller)